### DISCUSSION TITLE
Using grid as micro species of a another grid

### DISCUSSION THREAD

**Discussion by curiosity29** [gama-platform/gama#71] [Q&A] [answered]
I've read in the document that there's no difference between a grid to a normal species and a normal species is just a micro species of the world species. So I tried to define a grid inside a grid, and it showed no error but I can't access it or show it, it also doesn't create itself on initialize. It works fine with a micro species inside a grid so I don't know what's wrong with a micro grid. I was able to use a separate smaller grid for my problem but it feels a bit clunky.

**Comment by AlexisDrogoul** [comment, marked as answer]
It is surprising that you do not get a warning. Or even an error. Grids are a very special type of species, and they are supposed to be global (i.e. they can only reside in the simulation) and they should not be redefined inside other species. As a matter of fact, their agents are created automatically to cover all the surface of the environment, so there is no possibility to create "sub-grids" (at least using `grid`)

**Reply by curiosity29** [reply]
I get no error or warning even when running the simulation, it's just like that grid was never defined. Is there any other option for automatically dividing a specific shape from a species into smaller shapes instead of all the surface like a grid?

**Comment by AlexisDrogoul** [comment]
I think you can use the different versions of `to_rectangles` or `to_triangles` to tesselate the geometries of agents.

**Reply by curiosity29** [reply]
Thanks for the suggestion, I was able to divide my region into smaller rectangles with to_rectangles. It would be better if I can trim off the part where the small rectangle goes outside the original shape. There are some overlap functions but they seem to only return a boolean instead of geometry (where they overlap).
Also, are there any differences in showing the aspect of the micro species? It's doesn't seem to show the aspect like normally with:

aspect macro_aspect
{
    draw shape;
}
species micro
{
	aspect micro_aspect
	{
		draw shape;
	}
}
...
species macro aspect: macro_aspect;
species micro aspect: micro_aspect;

and I have to do it like this which seems a bit awkward:

aspect macro_aspect
{
	draw shape;
	loop mb over: members as list<micro>
	{ draw mb.shape at: mb.location; }
}
...
species macro aspect: macro_aspect;

**Reply by curiosity29** [reply]
Update: the function for overlapping geometry is `inter`. The micro species aspect still doesn't work though.

**Comment by sriramab** [comment]
Hi the small rectangles i think are points, you can try removing them like so with the area property (area of point =0)
`micro_species <- (to_rectangles(self.shape,{6,6}) where (each.area>0.01));`

**Reply by curiosity29** [reply]
Wow, I didn't know there was a third option which is exactly what I want where it trims off the part that goes outside the original shape. Although removing the very small one is probably a good idea.
I was using the format written in the documentation like this
`to_rectangles (geometry, point, bool) ---> list<geometry>`
Turn out there was another unmentioned use similar to your example
`to_rectangles (geometry, point) ---> list<geometry>`
